dask_kubernetes/core.py

```diff
@@ -181,67 +369,64 @@ def __init__(
             with open(fn) as f:
                 d = yaml.safe_load(f)
             d = dask.config.expand_environment_variables(d)
-            pod_template = make_pod_from_dict(d)
+            self.pod_template = make_pod_from_dict(d)
 
-        if not pod_template:
+        if not self.pod_template:
             msg = (
                 "Worker pod specification not provided. See KubeCluster "
                 "docstring for ways to specify workers"
             )
             raise ValueError(msg)
 
-        pod_template = clean_pod_template(pod_template)
-        ClusterAuth.load_first(auth)
+        self.pod_template = clean_pod_template(self.pod_template)
+        await ClusterAuth.load_first(self.auth)
 
         self.core_api = kubernetes.client.CoreV1Api()
 
-        if namespace is None:
-            namespace = _namespace_default()
+        if self._namespace is None:
+            self._namespace = _namespace_default()
 
-        name = name.format(
+        self._generate_name = self._generate_name.format(
             user=getpass.getuser(), uuid=str(uuid.uuid4())[:10], **os.environ
         )
-        name = escape(name)
-        self.pod_template = pod_template
+        self._generate_name = escape(self._generate_name)
 
         # Default labels that can't be overwritten
-        self.pod_template.metadata.labels["dask.org/cluster-name"] = name
+        self.pod_template.metadata.labels["dask.org/cluster-name"] = self._generate_name
         self.pod_template.metadata.labels["user"] = escape(getpass.getuser())
         self.pod_template.metadata.labels["app"] = "dask"
-        self.pod_template.metadata.labels["component"] = "dask-worker"
-        self.pod_template.metadata.namespace = namespace
-
-        self.cluster = LocalCluster(
-            host=host or socket.gethostname(),
-            scheduler_port=port,
-            n_workers=0,
-            **kwargs
-        )
+        self.pod_template.metadata.namespace = self._namespace
 
-        # TODO: handle any exceptions here, ensure self.cluster is properly
-        # cleaned up.
-        self.pod_template.spec.containers[0].env.append(
-            kubernetes.client.V1EnvVar(
-                name="DASK_SCHEDULER_ADDRESS", value=self.scheduler_address
-            )
-        )
-        if env:
+        if self.env:
             self.pod_template.spec.containers[0].env.extend(
                 [
                     kubernetes.client.V1EnvVar(name=k, value=str(v))
-                    for k, v in env.items()
+                    for k, v in self.env.items()
                 ]
             )
-        self.pod_template.metadata.generate_name = name
+        self.pod_template.metadata.generate_name = self._generate_name
+
+        finalize(
+            self,
+            self.sync,
+            _cleanup_pods,
```

Comment:
Now that this is an async def function I suspect that `finalize` won't work well.  Also, there is a chance that the event loop is now closed.  We might have to get a little creative here.  We probably have to have a function that uses the current event loop, if it's still active, or makes a new event loop if it isn't.  Alternatively, we could do this bit synchronously with the vanilla `kubernetes` API.

I may have handled this in the async dev branch.  I don't recall.

Reply:
Yeah I was noticing some warnings related to this. Vanilla k8s may be my preference.